### PR TITLE
Rename "useReviews" to "useReview" and use it in CT

### LINKED ISSUE #4800: Implement useReview mock
Implement mocked version of `useReview` in `commercetools` integration and theme.

Blocked by https://github.com/DivanteLtd/vue-storefront/issues/4775

### PR DESCRIPTION
**Pull request by filipsobol**
### Related Issues
Closes #4800

### Upgrade Notes and Changelog
- [ ] No upgrade steps required (100% backward compatibility and no breaking changes)
- [x] I've updated the [Upgrade notes](https://github.com/DivanteLtd/vue-storefront/blob/develop/docs/guide/upgrade-notes/README.md) and [Changelog](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md) on how to port existing Vue Storefront sites with this new feature

### CODE REVIEW

File: packages/core/core/src/types.ts [hunk 1]
```diff
@@ -253,6 +253,14 @@ export interface UserOrderGetters<ORDER, ORDER_ITEM> {
   [getterName: string]: (element: any, options?: any) => unknown;
 }
 
+export interface ReviewGetters<REVIEW> {
+  getReviewId: (review: REVIEW) => string;
+  getReviewAuthor: (review: REVIEW) => string;
+  getReviewMessage: (review: REVIEW) => string;
+  getReviewRating: (review: REVIEW) => number;
+  getReviewDate: (review: REVIEW) => string;
```
Comment (andrzejewsky) [inline packages/core/core/src/types.ts:250]: one missing: getTotalReviews => number, Also I'm considering a more advanced rating where you can see number of people per rate. eg: 

Screenshot 2020-09-09 at 11 56 33

so i Suggest following getters:

getAvererageRating => number - that gives ou the average rate
getTotalReviews => number - the total number of reviews
getReviewRating => AgnosticRating - that gives you more structured reviews data

```ts
interface AgnosticRate {
  rate: number;
  count: number;
}

interface AgnosticRating {
  averageRate: number; // same as getAvererageRating
  totalReviews: number; // same as getTotalReviews
  rates: Rate[]; // structured rating eg: { rate:5, count 123213 } etc
}
```

now when you want to only display the average rate you can use getAvererageRating but if you need something like on the screenshot above - getReviewRating that contains everything to display the rating. 

CC @filrak

**Review by andrzejewsky — CHANGES_REQUESTED** [review]

File: packages/core/core/src/types.ts (hunk 1 above)
Comment (filrak) [reply packages/core/core/src/types.ts:250]: I don;t think we need `interface AgnosticRate {
  rate: number;
  count: number;
}`

we could have just
```
 rates: { rate, count }
```

Also AgnosticRating -> AgnosticReviewsSummary ?

Comment (andrzejewsky) [reply packages/core/core/src/types.ts:250]: @filipsobol 

getAvererageRating => number - that gives ou the average rate
getTotalReviews => number - the total number of reviews
getReviewSummary => AgnosticRate[] - that gives you more structured reviews data

interface AgnosticRate { rate: number count: number }

**Review by andrzejewsky — APPROVED** [review]

**Review by filrak — APPROVED** [review]

File: packages/commercetools/api-client/src/api/addReview/index.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
+const addReview = () => {
```
[the remark this answers is not in the document]
Comment (filipsobol) [reply packages/commercetools/api-client/src/api/addReview/index.ts]: This is from some old version of this PR and no longer present.

File: packages/commercetools/api-client/src/api/getReview/index.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
+const getReview = () => {
```
[the remark this answers is not in the document]
Comment (filipsobol) [reply packages/commercetools/api-client/src/api/getReview/index.ts]: This is from some old version of this PR and no longer present.

File: packages/commercetools/composables/src/getters/reviewGetters.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  results: ReviewItem[];
+};
+
+type ReviewItem = Versioned & {
+  authorName: Scalars['String'];
+  text: Scalars['String'];
+  rating: Scalars['Int'];
+};
+
+export const getItems = (review: Review): ReviewItem[] => review?.results || [];
```
[the remark this answers is not in the document]
Comment (filipsobol) [reply packages/commercetools/composables/src/getters/reviewGetters.ts]: This is from some old version of this PR and no longer present.

**Review by filrak — APPROVED** [review]

**Review by filrak — CHANGES_REQUESTED** [review]
imho we can export all public api from `useReview.ts`since the alias is a single file

**Review by filrak — APPROVED** [review]

**Pull request merged**